### PR TITLE
End game view

### PR DESCRIPTION
**Pull request by joeleba**
- End Game View
- Asking for player's name:
  - Did it in first menu to ensure single player mode has player's name too
- Resolve `.PlayerLost` and `.GameDidEnd`

### CODE REVIEW

File: Courir/Courir/Constants.swift [hunk 1]
```diff
@@ -49,7 +49,8 @@ let obstacleNonFloatingTexture = SKTexture(imageNamed: "iso_non_floating_obstacl
 let obstacleFloatingTexture = SKTexture(imageNamed: "iso_floating_obstacle")
 
 // My Details
+var myName: String?
```
Comment (karen) [inline Courir/Courir/Constants.swift:52]: I think we can make use of `NSUserDefaults` for this (in the future(?))

File: Courir/Courir/LogicEngine.swift [hunk 2]
```diff
@@ -112,6 +123,13 @@ class LogicEngine {
         }
     }
     
+    // MARK: sending data
+    private func validToSendDataHandleEvent(player: Player) -> Bool {
```
Comment (karen) [inline Courir/Courir/LogicEngine.swift]: Can this method be less verbose? :sweat_smile:

File: Courir/Courir/MenuViewController.swift [hunk 3]
```diff
@@ -10,13 +10,21 @@ import UIKit
 import SpriteKit
 
 class MenuViewController: UIViewController {
-
+    private var nameTextField: UITextField?
```
Comment (karen) [inline Courir/Courir/MenuViewController.swift]: Do you have to save the text field and save action as instance variables? Can we just pass the variables into your validation functions? It seems kind of weird.

File: Courir/Courir/LogicEngine.swift (hunk 2 above)
Comment (joeleba) [inline Courir/Courir/LogicEngine.swift]: Done

File: Courir/Courir/MenuViewController.swift (hunk 3 above)
Comment (joeleba) [inline Courir/Courir/MenuViewController.swift]: Removed text field as instance variable. Save action needs to stay as instance variable for selector function `textFieldDidChange:` to work

**Pull request merged**